controller/lb_controller.go

```diff
@@ -0,0 +1,408 @@
+		// Two different versions of the same LoadBalancer will always have different RVs.
+		return
+	}
+
+	// can not change loadbalancer type from internal to external
+	if old.Type == lbapi.LoadBalancerTypeInternal && cur.Type == lbapi.LoadBalancerTypeExternal {
+		log.Warn("Forbidden to change the type of loadblancer, revert it", log.Fields{"from": old.Type, "to": cur.Type})
+		revert, err := lbc.clone(cur)
+		if err != nil {
+			return
+		}
+		revert.LoadBalancerSpec.Type = old.Type
+		_, err = lbc.tprClient.CaicloudV1beta1().LoadBalancers(cur.Namespace).Update(revert)
+		if err != nil {
+			log.Error("revert loadbalancer type error", log.Fields{"err": err})
+		}
+
+		return
+	}
+
+	log.Info("Updating LoadBalancer", log.Fields{"name": old.Name})
+	lbc.helper.Enqueue(cur)
+}
+
+func (lbc *LoadBalancerController) deleteLoadBalancer(obj interface{}) {
+	lb, ok := obj.(*lbapi.LoadBalancer)
+
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+		lb, ok = tombstone.Obj.(*lbapi.LoadBalancer)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a LoadBalancer %#v", obj))
+			return
+		}
+	}
+
+	log.Info("Deleting LoadBalancer", log.Fields{"name": lb.Name})
+
+	lbc.helper.Enqueue(lb)
+}
+
+func (lbc *LoadBalancerController) clone(lb *lbapi.LoadBalancer) (*lbapi.LoadBalancer, error) {
+	lbi, err := scheme.Scheme.DeepCopy(lb)
+	if err != nil {
+		log.Error("Unable to deepcopy loadbalancer", log.Fields{"lb.name": lb.Name, "err": err})
+		return nil, err
+	}
+
+	nlb, ok := lbi.(*lbapi.LoadBalancer)
+	if !ok {
+		nerr := fmt.Errorf("expected loadbalancer, got %#v", lbi)
+		log.Error(nerr)
+		return nil, err
+	}
+	return nlb, nil
+}
+
+// doLabelAndTaints delete label and taints in nodesToDelete
+// add label and taints in nodes
+func (lbc *LoadBalancerController) doLabelAndTaints(nodesToDelete []*apiv1.Node, desiredNodes *VerifiedNodes) {
+	// delete labels and taints from old nodes
+	for _, node := range nodesToDelete {
+		copy, _ := scheme.Scheme.DeepCopy(node)
+		copyNode := copy.(*apiv1.Node)
+
+		// change labels
+		for key := range desiredNodes.Labels {
+			delete(copyNode.Labels, key)
+		}
+
+		// change taints
+		// maybe taints are not found, reorganize will return error but it doesn't not matter
+		// taints will not be changed
+		_, newTaints, _ := taints.ReorganizeTaints(copyNode, false, nil, []apiv1.Taint{
+			{Key: lbapi.TaintKey},
+		})
+		copyNode.Spec.Taints = newTaints
+
+		labelChanged := !reflect.DeepEqual(node.Labels, copyNode.Labels)
+		taintChanged := !reflect.DeepEqual(node.Spec.Taints, copyNode.Spec.Taints)
+		if labelChanged || taintChanged {
+			log.Debug("Delete labels and taints from old nodes", log.Fields{
+				"node": node.Name,
+			})
+			_, err := lbc.kubeClient.CoreV1().Nodes().Update(copyNode)
```

Comment:
There is a tool to create patches: `k8s.io/apimachinery/pkg/util/strategicpatch`